modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/poll/PollingSourceWrapper.java

```diff
@@ -209,6 +211,12 @@ private void poll(SourceCallback<T, A> sourceCallback) {
     }
 
     withWatermarkLock(() -> {
+      if (shouldSkipPoll()) {
```

Comment:
why skip vs queue? This should be discussed in a spec.  What's the scheduler default behavior on this regard? Can/should this be configured on the scheduler directly?